src/internal/packager/sbom/catalog.go

```diff
@@ -70,30 +70,59 @@ func Catalog(componentSBOMs map[string]*types.ComponentSBOM, imgList []string, t
 	}
 	builder.jsonList = json
 
-	// Generate SBOM for each image
-	currImage := 1
-	for _, tag := range imgList {
-		builder.spinner.Updatef("Creating image SBOMs (%d of %d): %s", currImage, imageCount, tag)
+	// Generate SBOM for all images at once using goroutines
 
-		// Get the image that we are creating an SBOM for
-		img, err := utils.LoadOCIImage(tmpPaths.Images, tag)
-		if err != nil {
-			builder.spinner.Errorf(err, "Unable to load the image to generate an SBOM")
-			return err
-		}
+	// Create channel for letting us know when an image's SBOM is done
+	imageProgress := make(chan string, len(imgList))
 
-		jsonData, err := builder.createImageSBOM(img, tag)
-		if err != nil {
-			builder.spinner.Errorf(err, "Unable to create SBOM for image %s", tag)
-			return err
-		}
+	type errorWithMessage struct {
+		err     error
+		message string
+	}
```

Comment:
this might also be useful in message.go as a more common pattern

Reply:
made concurrency stuff reusable and put that struct in message.go